shared/AppInsightsCommon/src/Interfaces/ICorrelationConfig.ts

```diff
@@ -55,8 +55,14 @@ export interface ICorrelationConfig {
     excludeRequestFromAutoTrackingPatterns?: string[] | RegExp[];
 
     /**
-     * Provide a way to enrich dependencies logs with context at the beginning of api call.
+     * Provide a way to enrich dependencies logs with context at the beginning of ajax call.
      * Default is undefined.
      */
-     addContextOnRequests?: () => ICustomProperties;
+     addAjaxContext?: (xhr?: XMLHttpRequest) => ICustomProperties;
```

Comment:
Can we have this as 1 function that takes a new "request context" object as if we want to pass something more (like detected performance values) this would require a breaking change to the signature of this function.

The new "request context" object (interface) can contain all of the fields as optional (xhr, fetch request, fetch response, etc)